app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java

```diff
@@ -34,7 +37,32 @@ public void init() throws AlreadyLockedException, IOException {
       throw alreadyLockedException;
     }
 
-    NetworkTablesJNI.startClient("localhost", 1735);
+    NetworkTable.setClientMode();
+
+    ChangeListener<String> serverChangeListener = (observable, oldValue, newValue) -> {
+      if (newValue.matches("[1-9](\\d{1,3})?")) {
+        NetworkTable.setTeam(Integer.parseInt(newValue));
+      } else if (newValue.isEmpty()) {
+        NetworkTable.setIPAddress("localhost");
+      } else {
+        NetworkTable.setIPAddress(newValue);
+      }
+    };
+    AppPreferences.getInstance().serverProperty().addListener(serverChangeListener);
+
+    ChangeListener<Integer> portChangeListener = (observable, oldValue, newValue) -> {
+      NetworkTable.shutdown();
+      if (newValue == null || newValue <= 0 || newValue >= 65535) {
+        NetworkTable.setPort(NetworkTable.DEFAULT_PORT);
```

Comment:
this is bad UX, since it doesn't update the UI when it sets it to the default value. this should be encapsulated into the field instead. (new PortField(), portField.setDefault(NetworkTable.DEFAULT_PORT) or something)